src/esploader.ts

```diff
@@ -309,6 +192,19 @@ export class ESPLoader {
       this.transport.tracing = options.enableTracing;
     }
 
+    if (options.resetConstructors?.classicReset) {
+      this.resetConstructors.classicReset = options.resetConstructors?.classicReset;
```

Comment:
you already checked the variable exists, so you don't need to do

```
this.resetConstructors.classicReset = options.resetConstructors.classicReset;
```

Also, you can just use 

```js
    this.resetConstructors = {
      classicReset: (transport, resetDelay) => new ClassicReset(transport, resetDelay),
      customReset: (transport, sequenceString) => new CustomReset(transport, sequenceString),
      hardReset: (transport, usingUsbOtg) => new HardReset(transport, usingUsbOtg),
      usbJTAGSerialReset: (transport) => new UsbJtagSerialReset(transport),
      ...(options.resetConstructors || {})
    };
```